crates/sui-indexer/src/indexer.rs

```diff
@@ -126,18 +122,41 @@ impl Indexer {
             "primary".to_string(),
             config.checkpoint_download_queue_size,
         );
-
         executor.register(worker_pool).await?;
+        let (exit_sender, exit_receiver) = oneshot::channel();
+        executors.push((executor, exit_receiver));
+        exit_senders.push(exit_sender);
+
+        if config.sources.data_ingestion_path.is_none() {
```

Comment:
Does `data_ingestion_path` being none imply colocation or not colocation? We should start a separate executor when things are not colocated. A comment would help here. Thanks!

Reply:
data_ingestion_path being none automatically implies non-colocated setup